lib/ansible/modules/web_infrastructure/jenkins_build.py

```diff
@@ -0,0 +1,273 @@
+    u'artifacts': [], u'timestamp': 1520431274718,
+    u'previousBuild': {u'url': u'http://localhost:8080/job/test/1/', u'number': 1},
+    u'number': 2, u'id': u'2', u'keepLog': False, u'url': u'http://localhost:8080/job/test/2/',
+    u'result': u'SUCCESS', u'executor': None, u'duration': 172,
+    u'_class': u'org.jenkinsci.plugins.workflow.job.WorkflowRun', u'nextBuild': None,
+    u'fullDisplayName': u'test #2', u'estimatedDuration': 905}
+'''
+
+import traceback
+import time
+import uuid
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+try:
+    import jenkins
+    python_jenkins_installed = True
+except ImportError:
+    python_jenkins_installed = False
+
+
+class JenkinsBuild:
+
+    def __init__(self, module):
+        self.module = module
+
+        self.params = module.params.get('params')
+        self.name = module.params.get('name')
+        self.password = module.params.get('password')
+        self.token = module.params.get('token')
+        self.user = module.params.get('user')
+        self.jenkins_url = module.params.get('url')
+        self.wait_build = module.params.get('wait_build')
+        self.wait_build_timeout = module.params.get('wait_build_timeout')
+        self.build_token = module.params.get('build_token')
+        self.build_number = 1
+        self.timeout = module.params.get('timeout')
+        self.console_output = module.params.get('console_output')
+
+        self.server = self.get_jenkins_connection()
+
+        self.result = {
+            'build_info': {}
+        }
+
+    def get_jenkins_connection(self):
+        try:
+            if (self.user and self.password):
+                return jenkins.Jenkins(self.jenkins_url, self.user, self.password, self.timeout)
+            elif (self.user and self.token):
+                return jenkins.Jenkins(self.jenkins_url, self.user, self.token, self.timeout)
+            elif (self.user and not (self.password or self.token)):
+                return jenkins.Jenkins(self.jenkins_url, self.user, timeout=self.timeout)
+            else:
+                return jenkins.Jenkins(self.jenkins_url, timeout=self.timeout)
+        except Exception as e:
+            self.module.fail_json(msg='Unable to connect to Jenkins server, %s' % to_native(e),
+                                  exception=traceback.format_exc())
+
+    def job_exists(self):
+        try:
+            return bool(self.server.job_exists(self.name))
+        except Exception as e:
+            self.module.fail_json(msg='Unable to validate if job exists, %s for %s' % (to_native(e),
+                                  self.jenkins_url), exception=traceback.format_exc())
+
+    def wait_job_build(self):
+        for dummy in range(1, self.wait_build_timeout):
+            if self.server.get_build_info(self.name, self.build_number)['building']:
+                time.sleep(1)
+            else:
+                return
```

Comment:
It's possible to get a `jenkins.JenkinsException: job[your_job] number[1] does not exist` here. Maybe adding a try/catch for this?
```
            try:
                if self.server.get_build_info(self.name, self.build_number)['building']:
                    time.sleep(1)
                else:
                    return
            except jenkins.JenkinsException as e:
                time.sleep(1)
```